pandas/core/indexing.py

```diff
@@ -146,7 +146,8 @@ def _get_setitem_indexer(self, key):
             return self._convert_tuple(key, is_setter=True)
 
         axis = self.obj._get_axis(0)
-        if isinstance(axis, MultiIndex):
+
+        if isinstance(axis, MultiIndex) and not isinstance(key, int):
```

Comment:
also this should *only* be true if ``self.name == 'iloc'``, IOW we have a positional indexer.